Sources/PostgresNIO/Connection/PostgresConnection.swift

```diff
@@ -448,6 +448,27 @@ extension PostgresConnection {
             self.channel.write(task, promise: nil)
         }
     }
+
+    /// Execute a prepared statement, taking care of the preparation when necessary
+    public func execute<P: PreparedStatement>(
+        _ preparedStatement: P,
+        logger: Logger
+    ) async throws -> AsyncThrowingMapSequence<PostgresRowSequence, P.Row>
```

Comment:
also let's overload this method with a method where `Statement.Row == ()` where we only return the command tag?